modyn/supervisor/internal/eval/handler.py

```diff
@@ -83,56 +85,88 @@ def get_eval_requests_after_pipeline(self, df_trainings: pd.DataFrame) -> list[E
         assert {"trigger_id", "training_id", "id_model", "first_timestamp", "last_timestamp"} - (
             set(df_trainings.columns)
         ) == set()
+        df_trainings = df_trainings.copy()
```

Comment:
Any reason for this copy or leftover?

Reply:
because of call by reference. I wanted to avoid weird side effects at the caller's side. the training dataframe is usually < 1k lines, so performance wouldn't be a big concern.